### ISSUE TITLE
amp-bind: select[required] not respected in a form 

### ISSUE THREAD

**Issue by tmoffett-centrify** [ampproject/amphtml#17994]
## What's the issue?
Dynamically shown/hidden `select` form fields don't properly work with a dynamically calculated `required` attribute using `amp-bind`. The specific form field probably doesn't matter, but my specific issue was with a `select` field.

## How do we reproduce the issue?
This JSBin should show the exact issue and should be straightforward.
https://jsbin.com/yawivafiye/edit?html,output

1. There are two `select` inputs. First `select` makes the second one conditionally show up depending on selection.
2. The second `select` is generated by a mustache template in an `amp-list`.
3. The `amp-list` is wrapped in a `div` that gets shown/hidden based on the selection of the first `select`. It is hidden by default with custom CSS. Using the `[hidden]` attribute doesn't seem adequate since I need to have it hidden by default. 
4. Selecting `required` will make the second `select` show.
5. Selecting `required` from first `select` should also make the second `select` required but it doesn't.

This seems to only be an issue for this specific case of using the mustache template to dynamically generate this "special" content. I am able to have other fields dynamically be required and shown/hidden without issue.

## What browsers are affected?

Chrome 68.0.3440.106

## Which AMP version is affected?

Version 1536699515199

**Comment by ampprojectbot** [comment]
This issue hasn't been  updated in awhile. @choumx Do you have any updates?

**Comment by choumx** [comment]
Either I can't repro this or I don't understand the issue. 

amp-bind does set the `required` attribute on the `select` element as expected:

```html
<select data-amp-bind-required="something.value=='required'" i-amphtml-binding="" id="some-drop-down2" on="change:AMP.setState({someDropDownSelection:{value:event.value}})" required="" role="listitem" class="user-invalid valueMissing">
```

The form is not submitted when an empty value is selected on the `<select required>` element.

**Comment by choumx** [comment]
@tmoffett-centrify Please clarify the bug and I'll reopen the issue.

**Comment by tmoffett-centrify** [comment]
I think you missed the point of the issue. It had to do with dynamically showing a field as the main difference from a normal `select` field, so I'm not sure why you renamed the issue. I thought I came up with a name that properly described it. I put detailed steps/explanation and a JSBin too in my original issue.

Anyway, this might have been fixed in a recent update to AMP.  I tried my original JSBin example and it works now, but it wasn't working when I first posted the issue.
